mlflow/server/js/src/experiment-tracking/components/CompareRunView.tsx

```diff
@@ -155,542 +155,450 @@ export class CompareRunView extends Component<CompareRunViewProps, CompareRunVie
             return value;
           }
         } catch (e) {
-          return value;
+          return value; // Fallback if JSON parsing fails
         }
       },
     );
-    if (dataRows.length === 0) {
-      return (
-        <h2>
-          <FormattedMessage
-            defaultMessage="No parameters to display."
-            description="Text shown when there are no parameters to display"
-          />
-        </h2>
-      );
-    }
-    return (
+
+    return dataRows.length === 0 ? (
+      <h2>
+        <FormattedMessage defaultMessage="No parameters to display." description="No parameters to display" />
+      </h2>
+    ) : (
       <table
         className="table compare-table compare-run-table"
-        css={{ maxHeight: '500px' }}
-        onScroll={this.onCompareRunTableScrollHandler}
+        style={{ maxHeight: '500px' }}
+        onScroll={onCompareRunTableScrollHandler}
       >
         <tbody>{dataRows}</tbody>
       </table>
     );
-  }
+  };
 
-  renderPrettyJson(jsonValue: any) {
+  const renderPrettyJson = (jsonValue: any) => {
     return <pre>{JSON.stringify(jsonValue, null, 2)}</pre>;
-  }
+  };
 
-  renderMetricTable(colWidth: any, experimentIds: any) {
-    const dataRows = this.renderDataRows(
-      this.props.metricLists,
+  const renderMetricTable = (colWidth: number, experimentIds: string[]) => {
+    const dataRows = renderDataRows(
+      metricLists,
       colWidth,
-      // @ts-expect-error TS(4111): Property 'onlyShowMetricDiff' comes from an index ... Remove this comment to see the full error message
-      this.state.onlyShowMetricDiff,
+      onlyShowMetricDiff,
       false,
-      (key, data) => {
-        return (
-          <Link
-            to={Routes.getMetricPageRoute(
-              this.props.runInfos.map((info) => info.runUuid).filter((uuid, idx) => data[idx] !== undefined),
-              key,
-              experimentIds,
-            )}
-            title="Plot chart"
-          >
-            {key}
-            <i className="fas fa-chart-line" css={{ paddingLeft: '6px' }} />
-          </Link>
-        );
-      },
+      (key, data) => (
+        <Link
+          to={Routes.getMetricPageRoute(
+            runInfos.map((info) => info.runUuid).filter((uuid, idx) => data[idx] !== undefined),
+            key,
+            experimentIds,
+          )}
+          title="Plot chart"
+        >
+          {key}
+          <i className="fas fa-chart-line" style={{ paddingLeft: '6px' }} />
+        </Link>
+      ),
       Utils.formatMetric,
     );
-    if (dataRows.length === 0) {
-      return (
-        <h2>
-          <FormattedMessage
-            defaultMessage="No metrics to display."
-            description="Text shown when there are no metrics to display"
-          />
-        </h2>
-      );
-    }
-    return (
+
+    return dataRows.length === 0 ? (
+      <h2>
+        <FormattedMessage defaultMessage="No metrics to display." description="No metrics to display" />
```

Comment:
for the string changes, could we keep the `description` the same so they don't require retranslation?